From 5189442aad706583ab54b54fbce1fa3dff7ae87d Mon Sep 17 00:00:00 2001
From: Hideki Yamane <henrich@debian.org>
Date: Tue, 27 Feb 2018 16:27:14 +0900
Subject: [PATCH] allow both codename and script file as script parameter

As same as #742672, allow both codename and script file as script
parameter
---
 debootstrap | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/debootstrap b/debootstrap
index fcdb20f..3315d92 100755
--- a/debootstrap
+++ b/debootstrap
@@ -395,7 +395,7 @@ else
        if [ -z "$1" ] || [ -z "$2" ]; then
                usage_err 1 NEEDSUITETARGET "You must specify a suite and a target."
        fi
-       SUITE="$1"
+       SUITE="$(echo $1 | awk -F / '{ print $NF }')"
        TARGET="$2"
        USER_MIRROR="$3"
        TARGET="${TARGET%/}"
@@ -407,7 +407,14 @@ else
                fi
        fi
 
-       SCRIPT="$DEBOOTSTRAP_DIR/scripts/$1"
+       if [ -e "$DEBOOTSTRAP_DIR/scripts/$1" ]; then
+               SCRIPT="$DEBOOTSTRAP_DIR/scripts/$1"
+       elif [ -e "$1" ]; then
+               SCRIPT="$1"
+       else
+               echo "Please set proper codename"
+       fi
+              
        if [ -n "$VARIANT" ] && [ -e "${SCRIPT}.${VARIANT}" ]; then
                SCRIPT="${SCRIPT}.${VARIANT}"
                SUPPORTED_VARIANTS="$VARIANT"
-- 
2.16.2
